### PR TITLE
Update docs for payment module

### PR DESCRIPTION
**Pull request by rootulp**
## TODO
- [x] Split out changes that are only relevant post ADR 008 implementation and target feature branch
- [x] Document gas cost for a message
- [x] [Potentially] remove outdated code blocks from README.md

## Future PRs
- [ ] Params section needs to be updated after params are added to payment module https://github.com/celestiaorg/celestia-app/pull/893
- [ ] [Potentially] add client section (motivated by https://github.com/cosmos/cosmos-sdk/tree/main/x/auth#client)

### CODE REVIEW

File: x/payment/spec/docs.md [hunk 1]
```diff
@@ -1,117 +1,46 @@
-# Abstract
+# `x/payment`
```
Comment (rootulp) [inline x/payment/spec/docs.md:1]: A follow-up PR will move this file to `x/payment/README.md` per https://github.com/celestiaorg/celestia-app/issues/330#issuecomment-1303118116

File: x/payment/spec/docs.md [hunk 2]
```diff
@@ -1,117 +1,46 @@
...
 
 ## State
 
-- The sender’s account balance, via the bank keeper’s [`Burn`](https://github.com/cosmos/cosmos-sdk/blob/531bf5084516425e8e3d24bae637601b4d36a191/x/bank/spec/01_state.md) method.
-- The standard incrememnt of the sender's account number via the [auth module](https://github.com/cosmos/cosmos-sdk/blob/531bf5084516425e8e3d24bae637601b4d36a191/x/auth/spec/02_state.md).
-
-## Messages
-
-- [`MsgWirePayForData`](https://github.com/celestiaorg/celestia-app/blob/b4c8ebdf35db200a9b99d295a13de01110802af4/x/payment/types/tx.pb.go#L32-L40)
+The payment module doesn't maintain it's own state.
```
Comment (SweeXordious) [inline x/payment/spec/docs.md:11]:
```suggestion
The payment module doesn't maintain its own state.
```

**Review by SweeXordious — APPROVED** [review]
LGTM, deferring to Evan for final decision

File: x/payment/spec/docs.md (hunk 1 above)
Comment (rootulp) [reply x/payment/spec/docs.md:1]: https://github.com/celestiaorg/celestia-app/pull/1057

**Pull request merged**